ps-pxc-dist/apt_release.sh

```diff
@@ -0,0 +1,113 @@
+export GPG_TTY=$(tty)
+echo GPG_TTY is ${GPG_TTY}
+#
+DIRECTORIES=""
+
+# Create list of packages to push:
+for dir in $(printenv | grep _PATH | sed 's/^.*=//g'); do
+    DIRECTORIES+=" $dir"
+done
+
+if [ -z $DIRECTORIES ]; then
+    echo "No PATH specified..."
+    exit 1
+fi
+#
+REPOCOMP=$(echo "${COMPONENT}" | tr '[:upper:]' '[:lower:]')
+REPOPUSH_ARGS=""
+
+#if [[ ${REMOVE_BEFORE_PUSH} = true ]] || [[ ${COMPONENT} != RELEASE ]]; then
+if [[ ${REMOVE_BEFORE_PUSH} = true ]]; then
+    REPOPUSH_ARGS+=" --remove-package"
+fi
+#fi
+#
+#
+if [[ ${REPOSITORY} == "PDPXC" ]]; then
+  REPOPATH_TMP=repo-copy/pdpxc-${REPOSITORY_VERSION}/apt
+  if [ $REPOSITORY_VERSION_MAJOR == true ]; then
+    REPOPATH_TMP+=" repo-copy/pdpxc-$(echo ${REPOSITORY_VERSION} | awk -F"." 'BEGIN { OFS = "." }{ print $1, $2}')/apt"
+  fi
+  export PATH="/usr/local/reprepro5/bin:${PATH}"
+fi
+
+if [[ ${REPOSITORY} == "PDPS" ]]; then
+  REPOPATH_TMP=repo-copy/pdps-${REPOSITORY_VERSION}/apt
+  if [ $REPOSITORY_VERSION_MAJOR == true ]; then
+    REPOPATH_TMP+=" repo-copy/pdps-$(echo ${REPOSITORY_VERSION} | awk -F"." 'BEGIN { OFS = "." }{ print $1, $2}')/apt"
+  fi
+  export PATH="/usr/local/reprepro5/bin:${PATH}"
+fi
+#
+for REPOPATH in $REPOPATH_TMP; do
+    for $dir in $DIRECTORIES; do
+        #
+        set -e
+        echo "<*> reprepro binary is $(which reprepro)"
+        pushd /srv/UPLOAD/$dir/binary/debian
+        echo "Looking for Debian build directories..."
+        if [[ ${COMPONENT} != RELEASE ]]; then
+            CODENAMES=$(ls -1)
+        fi
+
+        echo Distributions are: ${CODENAMES}
+        popd
+        #
+        #######################################
+        # source pushing, it's a bit specific #
+        #######################################
+        #
+        if [[ ${REMOVE_LOCKFILE} = true ]]; then
+            echo "Removing lock file as requested..."
+            sudo rm -vf  ${REPOPATH}/db/lockfile
+        fi
+        #
+        if [[ ${REPOCOMP} = release ]]; then
+            export REPOCOMP=main
+        # pushing sources
+            cd /srv/UPLOAD/$dir/source/debian
```

Comment:
please add check if source directory is available.
There are some cases once we don't have it. So we should push packages without source packages